db/db_impl/db_impl_open.cc

```diff
@@ -34,6 +35,14 @@ Options SanitizeOptions(const std::string& dbname, const Options& src) {
 DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src) {
   DBOptions result(src);
 
+  if (result.file_system == nullptr) {
```

Comment:
Nothing wrong but I feel a little bit odd seeing comparing s shared_ptr to nullptr directly. Directly using `if (result.file_system)` feels more natural to me.

Reply:
I think it is OK because both `==` and `!` are overloaded: https://en.cppreference.com/w/cpp/memory/shared_ptr/operator_cmp